infra/terraform/modules/kops_spec/main.tf

```diff
@@ -0,0 +1,138 @@
+data "template_file" "etcd_member" {
```

Comment:
So to state the obvious, this will render the template string/file interpolating the variables and then create/update these resources as if you wrote them yourself, right?

(Sorry for the silly question, I'm not familiar with `template_file` and documentation just says "Renders a template from a file")

Reply:
Sorry, I didn't understand it at all :facepalm: . This will generate config files for kops, they'll be rendered and we can then copy them from the output variable.

Reply:
Yep, `template_file` just takes some input variables, interpolates them into a template file and makes the result available via the `data.template_file.$name.rendered` attribute. As everything kops needs to know is available from the `aws_vpc`, `cluster_dns` and `federated_identity` modules in `platform_base`, all kops resources (`ClusterSpec`, `InstanceGroup`) can be generated within terraform and then piped into kops: `terraform output kops_spec | kops create -f -`